src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/UserAcspMembership.java

```diff
@@ -42,10 +47,22 @@ public ResponseEntity<ResponseBodyPost> addAcspMember(
 
     @Override
     public ResponseEntity<AcspMembership> getAcspMembershipForAcspId(
-            @NotNull String xRequestId,
-            @Pattern(regexp = "^[a-zA-Z0-9]*$") String id
+            final String xRequestId,
+            final String id
     ) {
-        return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1146)
+      LOG.debug(String.format("%s: Retrieving RequestId and membership id (%s) ...", xRequestId, id));
+      if (Objects.isNull(id)){
```

Comment:
Is this null check needed? I think the super method this is implementing might handle it, probably worth checking